providers/ofrep/src/main/java/dev/openfeature/contrib/providers/ofrep/OfrepProvider.java

```diff
@@ -0,0 +1,125 @@
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Value;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.validator.routines.UrlValidator;
+
+/**
+ * OpenFeature provider for OFREP.
+ */
+@Slf4j
+public final class OfrepProvider implements FeatureProvider {
+
+    private static final String OFREP_PROVIDER = "ofrep";
+    private static final long DEFAULT_EXECUTOR_SHUTDOWN_TIMEOUT = 1000;
+
+    private final Resolver ofrepResolver;
+    private Executor executor;
+
+    public static OfrepProvider constructProvider() {
+        return new OfrepProvider();
+    }
+
+    /**
+     * Constructs an OfrepProvider with the specified options.
+     *
+     * @param options The options for configuring the provider.
+     * @return An instance of OfrepProvider configured with the provided options.
+     * @throws IllegalArgumentException if any of the options are invalid.
+     */
+    public static OfrepProvider constructProvider(OfrepProviderOptions options) {
+        if (!isValidUrl(options.getBaseUrl())) {
+            throw new IllegalArgumentException("Invalid base URL: " + options.getBaseUrl());
+        }
+
+        if (options.getHeaders() == null) {
+            throw new IllegalArgumentException("Headers cannot be null");
+        }
+
+        if (options.getTimeout() == null
+                || options.getTimeout().isNegative()
+                || options.getTimeout().isZero()) {
+            throw new IllegalArgumentException("Timeout must be a positive duration");
+        }
+
+        if (options.getProxySelector() == null) {
+            throw new IllegalArgumentException("ProxySelector cannot be null");
+        }
+
+        if (options.getExecutor() == null) {
+            throw new IllegalArgumentException("Executor cannot be null");
+        }
+
+        return new OfrepProvider(options);
+    }
+
+    private OfrepProvider() {
+        this(new OfrepProviderOptions.Builder().build());
+    }
+
+    private OfrepProvider(OfrepProviderOptions options) {
+        this.executor = options.getExecutor();
+        this.ofrepResolver = new Resolver(
+                options.getBaseUrl(),
+                options.getHeaders(),
+                options.getTimeout(),
+                options.getProxySelector(),
+                options.getExecutor());
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> OFREP_PROVIDER;
+    }
+
+    @Override
+    public void shutdown() {
+        if (executor instanceof ExecutorService) {
+            try {
+                ExecutorService executorService = (ExecutorService) executor;
+                executorService.shutdownNow();
```

Comment:
can call _shutdown_ first for graceful termination, see [shutdownAndAwaitTermination ](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html).